src/fheroes2/agg/agg_image.cpp

```diff
@@ -416,6 +416,31 @@ namespace
                             buttonFontColor );
     }
 
+    void createCampaignButtonSet( const int campaignSetIcnId, const std::array<const char *, 5> & texts )
+    {
+        int emptyButtonIcnID = 0;
+        switch ( campaignSetIcnId ) {
+        case ICN::GOOD_CAMPAIGN_BUTTONS:
+            emptyButtonIcnID = ICN::EMPTY_GOOD_BUTTON;
+            break;
+        case ICN::EVIL_CAMPAIGN_BUTTONS:
+            emptyButtonIcnID = ICN::EMPTY_EVIL_BUTTON;
+            break;
+        case ICN::POL_CAMPAIGN_BUTTONS:
+            emptyButtonIcnID = ICN::EMPTY_POL_BUTTON;
+            break;
+        default:
+            // Was a new set of buttons added?
+            assert( 0 );
+            break;
+        }
+
+        for ( size_t i = 0; i < texts.size(); ++i ) {
+            const size_t icnIndex = 2 * i;
+            fheroes2::getTextAdaptedButton( _icnVsSprite[campaignSetIcnId][2 * i], _icnVsSprite[campaignSetIcnId][icnIndex + 1], texts[i], emptyButtonIcnID );
```

Comment:
Please replace `2 * i` with the cached `icnIndex`.
```suggestion
            fheroes2::getTextAdaptedButton( _icnVsSprite[campaignSetIcnId][icnIndex], _icnVsSprite[campaignSetIcnId][icnIndex + 1], texts[i], emptyButtonIcnID );
```